bitcoin/Cargo.toml

```diff
@@ -40,7 +40,7 @@ units = { package = "bitcoin-units", version = "0.1.0", default-features = false
 base64 = { version = "0.21.3", optional = true }
 ordered = { version = "0.2.0", optional = true }
 # Only use this feature for no-std builds, otherwise use bitcoinconsensus-std.
-bitcoinconsensus = { version = "0.20.2-0.5.0", default-features = false, optional = true }
+bitcoinconsensus = { version = "0.105.0+25.1", default-features = false, optional = true }
```

Comment:
Why bother if it's EOL?

Reply:
The latest released version includes Taproot support (which means it implements all consensus rules). The old version that we were using did not.

While libbitcoinconsensus is EOL'd this will not actually have an effect on anybody until the consensus rules change.

Reply:
I don't think it's a good idea to have a feature we know we won't be able to support eventually. If there was an IO-free version of libbitcoinkernel then I'd consider it but even that is horribly complicated IMO.

Reply:
We should certainly encapsulate the `bitcoinconsensus` dep sufficiently that switching to a future `bitcoinkernel` dep will not be API-visible.

I then disagree that this feature would be something we "know we won't be able to support eventually". Getting an IO-free libbitcoinkernel with a similar API to libbitcoinconsensus is a matter of negotiating with Core, which admittedly is not the fastest or most flexible project, but it consists mainly of people who I know IRL and see multiple times a year at conferences.

Reply:
> We should certainly encapsulate the `bitcoinconsensus` dep sufficiently that switching to a future `bitcoinkernel` dep will not be API-visible.

Only once we split feature and dependency namespace and rename the feature to `verification`. I'd be open to it if I didn't have my doubts about the possibility of having IO-free libbitcoinkernel. It also puts stabilization to some later MSRV but I'm OK with that.

The problem with making it IO-free is you have to do conditional compilation. And if you have conditional compilation of a *library* then you have to somehow make sure you link against the correct one statically/dynamically and make sure you convey it to users properly. AFAIK the current OS/Rust infrastructure is not capable of dealing with this.

What I think could work is if libbitcoinkernel was two libraries - one without IO and the other with IO depending on the former one. We would then use existing capabilities of OS packaging (in case of dynamic linking) and cargo to resolve dependencies - just depend on what's needed.

Reply:
I think that if there was a "full libbitcoinkernel" and a "IO-free libbitcoinkernel" which was a strict subset, and we only used the subset, we could dynamically link to either one without harm. (And with static linking we have full control, we just link to whatever we've vendored in our repo.)

As for splitting feature and dependency namespace, it's an interesting observation that as long as we have a `bitcoinconsensus` optional dep, we can't remove it without a breaking change, even if we don't want users to actually use it directly. I guess we could initially have an optional dep, then later replace it with an explicit feature which just turns on `bitcoinkernel`. We'd wind up with a vestigial `bitcoinconsensus` feature in 1.0 but that's not the end of the world.

I think we should revisit this closer to 1.0. I could be convinced that we should just rip out `bitcoinconsensus` entirely and make people use it directly instead of rust-bitcoin. But I'm not convinced yet and I think we can do this "at the last minute" if that's what we decide to do.